src/code_80027D00.c

```diff
@@ -3600,7 +3600,7 @@ void func_8002F730(Player *player, Camera *camera, s8 arg2, s8 arg3) {
         func_8003F46C(player, &sp5C, &sp68, &spE8, &sp78, &spCC, &spC8, &spC4);
     }
     player->unk_074 = func_802ABE30(spCC, spC8, spC4, player->unk_110.unk3A);
-    if (((player->unk_000 & 0x4000) == 0x4000) && ((temp_v0 = gActiveScreenMode, (temp_v0 == 0)) || (temp_v0 == 2) || (temp_v0 == 1))) {
+    if (((player->unk_000 & 0x4000) == 0x4000) && (((gActiveScreenMode == 0)) || (gActiveScreenMode == 2) || (gActiveScreenMode == 1))) {
```

Comment:
The comma usually infers a separate if statement. With some statements between the ifs.

Since this doesn't match. Might be best to leave the m2c version for now. If you would like help matching it. Feel free to join the Discord server.